pkgs/applications/altcoins/namecoin.nix

```diff
@@ -1,20 +1,32 @@
-{ stdenv, db4, boost, openssl, qt4, qmake4Hook, miniupnpc, unzip, namecoind }:
+{ stdenv, fetchFromGitHub, pkgconfig, autoreconfHook, openssl, db4, boost,
+libevent, libtool, libuuid, qt4, protobuf, withGui }:
 
 with stdenv.lib;
-stdenv.mkDerivation rec {
 
-  name = "namecoin-${version}";
-  version = namecoind.version;
-  src = namecoind.src;
+stdenv.mkDerivation rec {
+  version = "nc0.13.0rc1";
+  name = "namecoin" + (toString (optional (!withGui) "d")) + "-" + version;
 
-  buildInputs = [ db4 boost openssl unzip qt4 qmake4Hook miniupnpc ];
+  src = fetchFromGitHub {
+    owner = "namecoin";
+    repo = "namecoin-core";
+    rev = version;
+    sha256 = "17zz0rm3js285w2assxp8blfx830rs0ambcsaqqfli9mnaik3m39";
+  };
 
-  qmakeFlags = [ "USE_UPNP=-" ];
+  nativeBuildInputs = [ pkgconfig autoreconfHook ];
+  buildInputs = [ openssl db4 boost libtool libuuid
+                  protobuf libevent ]
+                  ++ optionals withGui [ qt4 ];
```

Comment:
db4 and protobuf are [only needed for the GUI](https://github.com/namecoin/namecoin-core/blob/6e7eb2678e5a12a84a086e8079807094e0db9a74/doc/build-unix.md#dependencies)

There are also some other different dependencies between our versions.

Reply:
Sorry, db4 is needed in any case, but there is the optional libqrencode dependency